### PR TITLE
Remove `JaxModuleWrapper`

### LINKED ISSUE #1603: Remove need for JaxModuleWrapper
As mentioned in https://github.com/scverse/scvi-tools/issues/1564, if we find a way to default `training` without explicitly passing it in, this should be enough to move the necessary logic from `JaxModuleWrapper` to `JaxBaseModuleClass` and deprecate `JaxModuleWrapper` entirely. This interface will be much more consistent with `BaseModuleClass`.

### PR DESCRIPTION
**Pull request by adamgayoso**
Fixes #1603

All functionality was pushed into the `JaxBaseModuleClass`

### CODE REVIEW

File: scvi/module/base/_base_module.py [hunk 1]
```diff
@@ -533,8 +556,147 @@ def loss(self, *args, **kwargs) -> LossRecorder:
...
+        return self.train_state.params
+
+    @property
+    def state(self) -> FrozenDict[str, Any]:  # noqa: D102
+        self._check_train_state_is_not_none()
+        return self.train_state.state
+
+    def state_dict(self) -> Dict[str, Any]:
+        """Returns a serialized version of the train state as a dictionary."""
+        self._check_train_state_is_not_none
```
Comment (adamgayoso) [inline scvi/module/base/_base_module.py:613]:
```suggestion
        self._check_train_state_is_not_none()
```

File: scvi/module/_jaxvae.py [hunk 2]
```diff
@@ -88,8 +91,9 @@ def setup(self):
             "disp", lambda rng, shape: jax.random.normal(rng, shape), (self.n_input, 1)
         )
 
-    def __call__(self, z: jnp.ndarray, batch: jnp.ndarray, training: bool = False):
+    def __call__(self, z: jnp.ndarray, batch: jnp.ndarray, training=False):
```
Comment (justjhong) [inline scvi/module/_jaxvae.py:94]: missing typing

Comment (justjhong) [reply scvi/module/_jaxvae.py:94]: also shouldnt this default to None?

Comment (justjhong) [reply scvi/module/_jaxvae.py:94]: should probably have a tests for both cases of the merge param

File: scvi/module/base/_base_module.py [hunk 3]
```diff
@@ -402,20 +411,34 @@ def forward(self, *args, **kwargs):
...
     use the instance attribute ``self.training`` to appropriately modify
     the behavior of the model whether it is in training or evaluation mode.
     """
 
-    training: bool
+    def configure(self) -> None:
+        """Add necessary attrs."""
+        self.training = None
+        self.train_state = None
+        self.seed = 0
```
Comment (justjhong) [inline scvi/module/base/_base_module.py:439]: TODO to use scvi seed here?

File: scvi/module/base/_base_module.py [hunk 4]
```diff
@@ -533,8 +556,147 @@ def loss(self, *args, **kwargs) -> LossRecorder:
...
+            lambda x: jax.device_put(x, device), self.train_state
+        )
+        self.seed_rng = jax.device_put(self.seed_rng, device)
+        self._rngs = jax.device_put(self._rngs, device)
+
+    def _check_train_state_is_not_none(self):
+        if self.train_state is None:
+            raise RuntimeError("Train state is not set. Module has not been trained.")
+
+    def as_bound(self) -> JaxBaseModuleClass:
```
Comment (justjhong) [inline scvi/module/base/_base_module.py:639]: do we need this?

**Review by justjhong — APPROVED** [review]
looks pretty good

File: scvi/module/base/_base_module.py [hunk 5]
```diff
@@ -533,8 +556,147 @@ def loss(self, *args, **kwargs) -> LossRecorder:
...
+        Calls ``self._split_rngs()`` resulting in newly generated RNGs on
+        every reference to ``self.rngs``.
+        """
+        return self._split_rngs()
+
+    def _set_rngs(self):
+        """Creates RNGs split off of the seed RNG for each RNG required by the module."""
+        required_rngs = self.required_rngs
+        rng_keys = random.split(self.seed_rng, num=len(required_rngs) + 1)
+        self.seed, module_rngs = rng_keys[0], rng_keys[1:]
```
Comment (justjhong) [inline scvi/module/base/_base_module.py:585]:
```suggestion
        self.seed_rng, module_rngs = rng_keys[0], rng_keys[1:]
```

File: scvi/module/_jaxvae.py (hunk 2 above)
Comment (adamgayoso) [reply scvi/module/_jaxvae.py:94]: Done

File: scvi/module/base/_base_module.py (hunk 4 above)
Comment (adamgayoso) [reply scvi/module/base/_base_module.py:640]: I think it's fine to leave public and keep it, it will be helpful for debugging things

File: scvi/module/base/_base_module.py (hunk 3 above)
Comment (adamgayoso) [reply scvi/module/base/_base_module.py]: Done.

File: scvi/train/_trainingplans.py [hunk 6]
```diff
@@ -1159,6 +1159,7 @@ def jit_validation_step(
     def validation_step(self, batch, batch_idx):
         """Validation step for Jax."""
         self.module.eval()
+        assert self.module.training is False
```
Comment (adamgayoso) [inline scvi/train/_trainingplans.py:1162]:
```suggestion
```
